File: tests/testdata/codegen/deferred-outputs-pp/dotnet/deferred-outputs.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;

return await Deployment.RunAsync(() => 
{
    var secondPasswordLength = new Pulumi.DeferredOutput<int>();
    var first = new Components.First("first", new()
    {
        PasswordLength = secondPasswordLength.Output,
    });

    var second = new Components.Second("second", new()
    {
        PetName = first.PetName,
    });

    secondPasswordLength.Resolve(second.PasswordLength);
    var loopingOverMany = new Pulumi.DeferredOutput<List<int>>();
    var another = new Components.First("another", new()
    {
        PasswordLength = loopingOverMany.Output.Apply(loopingOverMany => loopingOverMany.Length),
    });

    var many = new List<Components.Second>();
    for (var rangeIndex = 0; rangeIndex < 10; rangeIndex++)
    {
        var range = new { Value = rangeIndex };
        many.Add(new Components.Second($"many-{range.Value}", new()
        {
            PetName = another.PetName,
        }));
    }
    loopingOverMany.Resolve(Output.Create(many.Select((value, i) => new { Key = i.ToString(), Value = pair.Value }).Select(v => 
    {
        return v.PasswordLength;
    }).ToList()));
});

File: tests/testdata/codegen/deferred-outputs-pp/go/deferred-outputs.go
package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		secondPasswordLength, resolveSecondPasswordLength := pulumi.DeferredOutput(ctx)
		first, err := NewFirst(ctx, "first", &FirstArgs{
			PasswordLength: secondPasswordLength,
		})
		if err != nil {
			return err
		}
		second, err := NewSecond(ctx, "second", &SecondArgs{
			PetName: first.PetName,
		})
		if err != nil {
			return err
		}
		resolveSecondPasswordLength(second.PasswordLength)
		loopingOverMany, resolveLoopingOverMany := pulumi.DeferredOutput(ctx)
		another, err := NewFirst(ctx, "another", &FirstArgs{
			PasswordLength: float64(loopingOverMany.ApplyT(func(loopingOverMany []int) (pulumi.Int, error) {
				return pulumi.Int(len(loopingOverMany)), nil
			}).(pulumi.IntOutput)),
		})
		if err != nil {
			return err
		}
		var many []*Second
		for index := 0; index < 10; index++ {
			key0 := index
			_ := index
			__res, err := NewSecond(ctx, fmt.Sprintf("many-%v", key0), &SecondArgs{
				PetName: another.PetName,
			})
			if err != nil {
				return err
			}
			many = append(many, __res)
		}
		resolveLoopingOverMany("TODO: For expression")
		return nil
	})
}
